pkg/models/kr/keyrange.go

```diff
@@ -35,71 +40,189 @@ type KeyRange struct {
+		fallthrough
+	case qdb.ColumnTypeVarchar:
+		kr.LowerBound[attribInd] = string(raw)
 	}
+}
 
-	return len(kr) < len(other)
+func (kr *KeyRange) OutFunc(attribInd int) []byte {
+	switch kr.ColumnTypes[attribInd] {
+	case qdb.ColumnTypeInteger:
+		raw := make([]byte, 8)
+		_ = binary.PutVarint(raw, kr.LowerBound[attribInd].(int64))
+		return raw
+	case qdb.ColumnTypeUinteger:
+		raw := make([]byte, 8)
+		_ = binary.PutUvarint(raw, kr.LowerBound[attribInd].(uint64))
+		return raw
+	case qdb.ColumnTypeVarcharDeprecated:
+		fallthrough
+	case qdb.ColumnTypeVarchar:
+		return []byte(kr.LowerBound[attribInd].(string))
+	}
+	return nil
 }
 
-// CmpRangesEqual compares two byte slices, kr and other, and returns true if they are equal.
-// It checks if the lengths of kr and other are the same, and then compares their string representations.
-//
-// Parameters:
-//   - kr: The first byte slice to compare.
-//   - other: The second byte slice to compare.
-//
-// Returns:
-//   - bool: True if kr and other are equal, false otherwise.
-//
+func (kr *KeyRange) SendFunc(attribInd int) string {
+	switch kr.ColumnTypes[attribInd] {
+	case qdb.ColumnTypeInteger:
+		fallthrough
+	case qdb.ColumnTypeUinteger:
+		return fmt.Sprintf("%v", kr.LowerBound[attribInd])
+	default:
+		return fmt.Sprintf("'%v'", kr.LowerBound[attribInd])
+	}
+}
+
+func (kr *KeyRange) Raw() [][]byte {
+	res := make([][]byte, len(kr.ColumnTypes))
+
+	for i := 0; i < len(kr.ColumnTypes); i++ {
+		res[i] = kr.OutFunc(i)
+	}
+
+	return res
+}
+
+func (kr *KeyRange) SendRaw() []string {
+	res := make([]string, len(kr.ColumnTypes))
+
+	for i := 0; i < len(kr.ColumnTypes); i++ {
+		res[i] = kr.SendFunc(i)
+	}
+
+	return res
+}
+
+// TODO: use it
+var MissTypedKeyRange = fmt.Errorf("key range bound is mistyped")
+
 // TODO : unit tests
-func CmpRangesEqual(kr []byte, other []byte) bool {
-	if len(kr) == len(other) {
-		return string(kr) == string(other)
+func CmpRangesLess(bound KeyRangeBound, key KeyRangeBound, types []string) bool {
+	// Here we panic if we failed to convert key range bound
+	// element to expected type. We consider panic as much better
+	// result that data corruption caused by erroreus routing logic.
+	// Big TODO here is to use and check specific error of types mismatch.
+
+	for i := 0; i < len(bound); i++ {
+		switch types[i] {
+		case qdb.ColumnTypeInteger:
+			i1 := bound[i].(int64)
+			i2 := key[i].(int64)
+			if i1 == i2 {
+				// continue
+			} else if i1 < i2 {
+				return true
+			} else {
+				return false
+			}
+		case qdb.ColumnTypeVarchar:
+			i1 := bound[i].(string)
+			i2 := key[i].(string)
+			if i1 == i2 {
+				// continue
+			} else if i1 < i2 {
+				return true
+			} else {
+				return false
+			}
+		case qdb.ColumnTypeVarcharDeprecated:
+			i1 := bound[i].(string)
+			i2 := key[i].(string)
+			if i1 == i2 {
+				// continue
+			} else if CmpRangesLessStringsDeprecated(i1, i2) {
+				return true
+			} else {
+				return false
+			}
+		default:
+			// wtf?
```

Comment:
```suggestion
		default:
```
Comment can be dropped